src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -458,6 +458,7 @@ LedgerEntryTypesMatch::visitEntry(
         switch (after->getType())
         {
             case ltACCOUNT_ROOT:
+            case ltACCOUNT_PERMISSION:
```

Comment:
I can't leave a comment on unchanged lines so commenting here for ValidClawback::finalize:924, which is this line:

     AccountID const issuer = tx.getAccountID(sfAccount);

The `issuer` in this case could be either `sfAccount` or `sfOnBehalfOf`, right?

Reply:
This is missed out. I'll also check all the other similar places which is missed out.

Reply:
@gregtatcam 
I also need to make change to AcceptedLedgerTx.cpp, LedgerToJson.cpp, and NetworkOPs.cpp, because they also check the account id when the transaction type is ttOFFER_CREATE. I'm working on this as well